darts/utils/data/simple_inference_dataset.py

```diff
@@ -92,14 +100,19 @@ def __getitem__(self, idx: int) -> Tuple[TimeSeries, Optional[TimeSeries], Optio
             if self.n > self.output_chunk_length:
 
                 # check that enough future covariates are available
+                # block models need `n - output_chunk_length`
+                # recurrent models need `n`
                 last_required_future_covariate_ts = (
-                    target_series.end_time() + (self.n - self.output_chunk_length) * target_series.freq()
+                    target_series.end_time() + (self.n - self.output_chunk_length * (1 - int(self.model_is_recurrent)))
+                    * target_series.freq()
                 )
+                req_cov_string = 'n' if self.model_is_recurrent else 'n - output_chunk_length'
                 raise_if_not(covariate_series.end_time() >= last_required_future_covariate_ts,
-                             'All covariates must be known `n - output_chunk_length` time steps into the future')
+                             'All covariates must be known `{}` time steps into the future'.format(req_cov_string))
 
                 # isolate necessary future covariates and add them to array
-                cov_future = covariate_series.drop_before(first_pred_time - covariate_series.freq())
-                cov_future = cov_future[:self.n - self.output_chunk_length]
+                cov_future = covariate_series.drop_before(first_pred_time - (1 - int(self.model_is_recurrent))
```

Comment:
How come the non-recurrent models need to be split one step before?

Reply:
In the shifted dataset, for recurrent models, I shift the covariates forward relative to the input series by one time step. This ensures that recurrent models have as input the most recent covariates when making a prediction (covariates with the same timestamp as the target). Because the RNN is trained that way, this shift has to be incorporated into the SimpleInferenceDatqset class as well, and this applies to future covariates too. This translates to this different cutoff point. You think there might be a better way of handling this? I do see that having two different ways of creating cov_future might not be ideal if someone is trying to understand the code.

Reply:
Thanks for the explanation! I think you can leave it like that, but maybe copy-paste what you wrote here into a comment in the code for our future selves :) thanks!

Reply:
Done